Class-9-(Exceptions-&-Build-Tools).md

```diff
@@ -0,0 +1,33 @@
+# Class 9: Exceptions and Build Tools
+
+## In the prep work this week, the students learned
+
+1. What an exception is and how to handle an exception.
+1. Common exceptions in C#
+1. What MSBuild is.
+1. How Visual Studio manages packages with NuGet.
+1. How to use debugging tools in Visual Studio.
+
+## Announcements
+
+1. Remind students that Assignment #3 is open for them to start working on it!
+1. Check with your course manager for any other important announcements.
+
+## Large Group Time (Instructor)
+
+### Class 9 Topics That Require Careful Attention
+
+1. Much of the reading is based around debugging and handling errors. Don't be afraid to share your own experiences with debugging and your own process when it comes to using debugging tools and exception handling.
+1. Share an exception example and go over the following:
+   1. How you decided which exception to use
+   1. How you decided what code to put in ``try`` statement and what code to put in ``catch`` statement.
+   1. Does it include a ``finally`` statement? Why or why not?
+1. Review MSBuild and NuGet. Go over what each one does and why each one matters.
+1. Show how to use debugging tools in Visual Studio to see the *Call Stack*, add breakpoints, and address common problems.
+
+## Small Group Time: Class 9 Studio (TA Notes)
+
+1. Students will be sharing their errors and trying to use debugging tools on ``HelloASPDotNETDemo``. Some things to keep in mind:
+   1. Don't be afraid to share your own stories of bugs and errors in your code! Emphasize that every programmer has to deal with bugs and that it gets a lot easier once you have a strategy in place and the tools 
+   1. Ask students why! Helping students figure out a process by which they can debug their code is the point of the studio.
```

Comment:
I don't quite get this line - what are the TAs asking students about?

Reply:
Clarified!